app/src/services/datasetService.js

```diff
@@ -63,33 +64,43 @@ class DatasetService {
+                                alertType: datasetQuery.type,
+                                alertBeginDate: datasetQuery.lastSentDate.toISOString().slice(0, 10),
+                                alertEndDate: new Date().toISOString().slice(0, 10),
+                                alertResult: result.data[0].value
+                            };
+
+                            // Execute EMAIL notification - sending an email
                             if (subscription.resource.type === 'EMAIL') {
-                                const areaName = subscription.params.area ? await DatasetService.getAreaName(subscription.params.area) : '';
-                                const datasetName = metadata[0].attributes.info.name || dataset.attributes.name || dataset.name;
-                                const data = {
-                                    subject: subscription.name ? subscription.name : `${datasetQuery.type} in ${areaName} above ${datasetQuery.threshold}`,
-                                    datasetName,
-                                    datasetId: datasetQuery.id,
-                                    datasetSummary: metadata[0].attributes.info.functions,
-                                    areaId: subscription.params.area ? subscription.params.area : '',
-                                    areaName,
-                                    alertName: subscription.name ? subscription.name : `${datasetQuery.type} in ${areaName} above ${datasetQuery.threshold}`,
-                                    alertType: datasetQuery.type,
-                                    alertBeginDate: datasetQuery.lastSentDate.toISOString().slice(0, 10),
-                                    alertEndDate: new Date().toISOString().slice(0, 10),
-                                    alertResult: result.data[0].value
-                                };
                                 logger.debug('Sending mail with data', data);
                                 let template = 'dataset-rw';
                                 if (subscription.env && subscription.env !== 'production') {
                                     template += `-${subscription.env}`;
                                 }
                                 MailService.sendMail(template, data, [{ address: subscription.resource.content }], 'rw'); // sender='rw'
+                            }
 
-                            } else {
-                                // @TODO resource.type === 'WEBHOOK'?
+                            // Execute URL notification - POSTing to webhook
+                            if (subscription.resource.type === 'URL') {
+                                // POST to URL configured in subscription.resource.content
+                                request.post(subscription.resource.content, { json: data }, (error, res, body) => {
```

Comment:
I'd like to see some validation that `subscription.resource.content` is an URL or, in case it's not, a better handling of that error

Reply:
Improved error handling, now diffing between cases when the POST request returns non-200 status codes and other errors (such as invalid URL).

However, I don't think there's much more to do here, since we have a catch below (https://github.com/gfw-api/gfw-subscription-api/blob/6751c6e28371f726bb9c13cca9295fb5d9312c24/app/src/services/datasetService.js#L106) that catches all errors thrown here. I added logs to help debug any problem that might arise!